src/scraper.py

```diff
@@ -43,75 +43,100 @@ def add_filetype(file_path: str):
+    elif (val in ['2mp','4mp','6mp','8mp','10mp','12mp','15mp','20mp','40mp','70mp']):
+        return key + 'lt%2Cislt:' + val
     else:
         return ""
 
+def process_image_aspectratio(val: str):
+    key = 'iar:'
+    if (val == 'tall'):
+        return key + 't'
+    elif (val == 'square'):
+        return key + 's'
+    elif (val == 'wide'):
+        return key + 'w'
+    elif (val == 'panoramic'):
+        return key + 'xw'
+
 def process_image_color(val: str):
-    if (val == "grayscale"):
+    if (val == "color"):
+        return "ic:color"
+    elif (val == "grayscale"):
         return "ic:gray"
     elif (val == "transparent"):
         return "ic:trans"
-    elif (val == "red"):
-        return "ic:specific%2Cisc:red"
-    elif (val == "orange"):
-        return "ic:specific%2Cisc:orange"
-    elif (val == "yellow"):
-        return "ic:specific%2Cisc:yellow"
-    elif (val == "green"):
-        return "ic:specific%2Cisc:green"
-    elif (val == "teal"):
-        return "ic:specific%2Cisc:teal"
-    elif (val == "blue"):
-        return "ic:specific%2Cisc:blue"
-    elif (val == "purple"):
-        return "ic:specific%2Cisc:purple"
-    elif (val == "pink"):
-        return "ic:specific%2Cisc:pink"
-    elif (val == "white"):
-        return "ic:specific%2Cisc:white"
-    elif (val == "gray"):
-        return "ic:specific%2Cisc:gray"
-    elif (val == "black"):
-        return "ic:specific%2Cisc:black"
-    elif (val == "brown"):
-        return "ic:specific%2Cisc:brown"
+    elif (val in ['red','orange','yellow','green','teal','blue','purple','pink','white','gray','black','brown']):
+        return "ic:specific%2Cisc:" + val
     else:
         return ""
 
 def process_image_type(val: str):
-    if (val == "clipart"):
-        return "itp:clipart"
-    elif (val == "lineart"):
-        return "itp:lineart"
-    elif (val == "animated"):
-        return "itp:animated"
+    if (val in ['face', 'photo', 'clipart', 'lineart', 'animated']):
+        return 'itp:' + val
     else:
         return ""
 
+def process_image_region(val: str):
+    if (val == ''):
+        return ''
+    else:
+        return 'ctr:country' + val.upper()
+
+def process_image_filetype(val: str):
+    if (val in ['jpg', 'gif', 'png', 'bmp', 'svg', 'webp', 'ico', 'raw']):
+        return 'ift:' + val
+
+def process_image_usage(val: str):
+    key = 'sur:'
+    if (val == 'cc'):
+        return key + 'cl'
+    elif (val == 'other'):
+        return key + 'ol'
+    else:
+        return ''
+
 def process_safesearch(val: str):
-    if (val == "on"):
-        return "on"
-    elif (val == "off"):
-        return "off"
+    if (val in ["on", "off"]):
+        return val
     else:
         return ""
 
 
-def setup_url(searchurl: str, imgsize: str, imgcolor: str, imgtype: str, safesearch: str):
+def setup_url(searchurl: str, imgsize: str, imgaspectratio: str, imgcolor: str, imgtype: str, imgregion: str, imgfiletype: str, imgusage: str, safesearch: str):
```

Comment:
The filters should be passed around as a single data structure. Long function definitions are boring and hard to read